packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -124,12 +125,16 @@ export function useVirtualizer<T extends object, V, W>(props: VirtualizerOptions
   // is up to the implementation using Virtualizer since we don't have refs
   // to all of the item DOM nodes.
   let lastFocusedKey = useRef(null);
+  let isFocusWithin = useRef(false);
   useEffect(() => {
     if (virtualizer.visibleRect.height === 0) {
       return;
     }
 
-    if (focusedKey !== lastFocusedKey.current) {
+    // Only scroll the focusedKey into view if the modality is not pointer to avoid jumps in position when clicking/pressing tall items.
+    // Exception made if focus isn't within the virtualizer (e.g. opening a picker via click should scroll the selected item into view)
+    let modality = getInteractionModality();
+    if (focusedKey !== lastFocusedKey.current && (modality !== 'pointer' || !isFocusWithin.current)) {
```

Comment:
An alternative to these Virtualizer changes would be to modify the following https://github.com/adobe/react-spectrum/blob/d764bb0570b011bca731b7da1e8f87e2f34b6e32/packages/%40react-spectrum/card/src/CardView.tsx#L106-L111
to have similar pointer modality checking logic to prevent `virtualizer.scrollToItem` from being called. This would then have to replicated in other places like ListBox where deemed necessary...

Reply:
I think the above logic of checking the focused item and the type of not pointer works. That is clearer for me to understand than your alternative.